test/helpers/utils.go

```diff
@@ -522,17 +532,16 @@ func DoesNotRunOn419Kernel() bool {
 	return !RunsOn419Kernel()
 }
 
-// RunsOnNetNextOr419Kernel checks whether a test case is running on the net-next
-// kernel (depending on the image, it's the latest kernel either from net-next.git
-// or bpf-next.git tree), or on the > 4.19.57 kernel.
-func RunsOnNetNextOr419Kernel() bool {
-	return RunsOnNetNextKernel() || RunsOn419Kernel()
+// RunsOn419OrLaterKernel checks whether a test case is running on the bpf-next,
+// 4.19.x (x > 57), or 5.4 kernels.
+func RunsOn419OrLaterKernel() bool {
+	return RunsOnNetNextKernel() || RunsOn419Kernel() || RunsOn54Kernel()
 }
 
-// DoesNotRunOnNetNextOr419Kernel is the complement function of
-// RunsOnNetNextOr419Kernel.
-func DoesNotRunOnNetNextOr419Kernel() bool {
-	return !RunsOnNetNextOr419Kernel()
+// DoesNotRunOn419OrLaterKernel is the complement function of
+// RunsOn419OrLaterKernel.
+func DoesNotRunOn419OrLaterKernel() bool {
```

Comment:
Yesterday I was looking at this part of the code. I was a little bit confused about the name of this function when I read. IMHO it should be DoesNotRunOn419AndLaterKernel or NotRunOn419OrLaterKernel. Does it make sense? I dont know if I understand correctly.

Reply:
If you look at `RunsOnNetNextOr419Kernel` definition we can generalize as `p OR q`. When you put a `!RunsOnNetNextOr419Kernel` it becomes a `¬ p AND ¬ q`, and not `¬ p OR ¬ q`. That was my point. Maybe I didn't get it right

Reply:
You forgot the parenthesis ;) To me it looks like `...RunOn..And...` means running (or not running, in our case) on _several_ kernels at the same time, which does not make much sense. I'd rather keep the `Or`, personally.

Reply:
Yes! Thank you for the explanation! I got the point :smile: